Gruntfile.js

```diff
@@ -1099,58 +988,314 @@ module.exports = function(grunt) {
+  // =========================================================================
+
   registerTask(grunt, 'build_chrome_app', [
     'base',
-    'copy:chrome_app'
+    'chromeAppMainCoreEnv',
+    'chromeAppExtMissing',
+    'copy:chromeApp',
+  ]);
+  registerTask(grunt, 'chromeAppMainCoreEnv',
+      'Builds build/dist/chrome/app/scripts/main.core-env.static.js', [
+    'compileTypescript',
+    'browserify:chromeAppMainCoreEnv'
+  ]);
+  registerTask(grunt, 'chromeAppExtMissing',
+      'Builds build/src/chrome/app/polymer/vulcanized.{html,js}', [
+    'compileTypescript',
+    'copy:resources',
+    'copy:chromeAppBower',
   ].concat(fullyVulcanize('chrome/app/polymer', 'ext-missing', 'vulcanized')));
+  
+  grunt.config.merge({
+    browserify: {
+      chromeAppMainCoreEnv: Rule.browserify('chrome/app/scripts/main.core-env'),
+    },
+    copy: {
+      chromeApp: Rule.copyLibs({
+        npmLibNames: ['freedom-for-chrome', 'forge-min'],
+        pathsFromDevBuild: [
+          'generic_core'
+        ].concat(backendFreedomModulePaths),
+        pathsFromThirdPartyBuild: backendThirdPartyBuildPaths,
+        files: getExtraFilesForCoreBuild(chromeAppDevPath).concat({
+          expand: true, cwd: 'src/',
+          src: ['icons/128_online.png', 'fonts/*'],
+          dest: chromeAppDevPath
+        }),
+        localDestPath: 'chrome/app/'
+      }),
+      chromeAppBower: {
+        expand: true, cwd: 'third_party',
+        src: ['bower/**/*'],
+        dest: chromeAppDevPath
+      },
+    }
+  });
+
+  // =========================================================================
+  // Chrome Extension
+  // =========================================================================
+
   registerTask(grunt, 'build_chrome_ext', [
     'base',
-    'copy:chrome_extension',
-    'copy:chrome_extension_additional',
-    'browserify:chromeExtMain',
-    'browserify:chromeContext'
-  ].concat(fullyVulcanize('chrome/extension/generic_ui/polymer', 'root', 'vulcanized', true)));
-
-  registerTask(grunt, 'build_chrome', [
-    'build_chrome_app',
-    'build_chrome_ext'
+    'chromeExtBackground',
+    'chromeExtContext',
+    'chromeExtRoot',
+    'copy:chromeExt',
+    'copy:chromeExtAdditional',
+  ]);
+  registerTask(grunt, 'chromeExtBackground', 
+      'Builds build/src/chrome/extension/scripts/background.static.js', [
+    'compileTypescript',
+    'browserify:chromeExtBackground'
   ]);
+  registerTask(grunt, 'chromeExtContext', 
+      'Builds build/src/chrome/extension/scripts/context.static.js', [
+    'compileTypescript',
+    'browserify:chromeExtContext'    
+  ]);
+  registerTask(grunt, 'chromeExtRoot',
+      'Builds build/src/chrome/extension/generic_ui/polymer/vulcanized.{html,static.js}', [
+    'compileTypescript',
+    'copy:resources',
+    'copy:chromeExt',
+    'copy:chromeExtAdditional',
+  ].concat(fullyVulcanize('chrome/extension/generic_ui/polymer', 'root', 'vulcanized', true)));
+  
+  grunt.config.merge({
```

Comment:
Interesting. `grunt.config.merge` essentially allows you co-locate task definitions. We haven't done that before but I can see a readability benefit.